src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2181,6 +2221,10 @@ std::error_code Session::receive_ident_message(SaltedFileIdent client_file_ident
         REALM_ASSERT_EX(m_last_version_selected_for_upload == 0, m_last_version_selected_for_upload);
 
         get_transact_reporter()->report_sync_transact(client_reset_old_version, client_reset_new_version);
+
+        if (has_pending_client_reset) {
```

Comment:
I don't think it is correct to clear the pending reset state here because the reset has just finished locally and it hasn't received any acknowledgement from the server yet.

Reply:
this doesn't necessarily clear anything, it schedules the async waiting for the server to ack all uploads and tell the client when there are no more downloads, which will then clear the reset state.

Reply:
Btw, the reason we check if there's a pending client reset and then only call this if there is one is so that we don't need to open a separate read transaction after calling `get_status()` if there are no client resets to do this dance with.

Reply:
Ah, got it thanks! I hadn't understood that `async_wait_for` is initiating an upload/download completion handler.